pkg/kvstore/store/syncstore.go

```diff
@@ -0,0 +1,190 @@
+// SPDX-License-Identifier: Apache-2.0
```

Comment:
Probably we want to have some metrics to indicate number of elements in workqueue + how often processing item failed.

Reply:
I personally agree. I'm wondering though whether to add those in a followup PR, since the clustermesh-apiserver is currently missing the glue logic to expose any metrics.

Reply:
https://github.com/cilium/cilium/pull/25260/commits/90b759df73db5292a5beb6d8e9efe91991708a5f introduces a new `cilium_kvstore_sync_queue_size` metric to indicate the current size of the queue (per `resource` and `source_cluster`):

```
cilium_kvstore_sync_queue_size{scope="services/v1",source_cluster="left"} 0
```

The number of failures, instead, is already available through the existing `kvstore_operations_duration_seconds` metrics (which is an histogram labeled by `scope`, `action`, `kind` and `outcome`).

Both metrics require https://github.com/cilium/cilium/pull/25316 to be exposed by the clustermesh-apiserver.